boards/ruiside/art_pi/art_pi.dts

```diff
@@ -0,0 +1,417 @@
+			gpios = <&gpioc 15 GPIO_ACTIVE_LOW>;
+			label = "USER LED1";
+		};
+
+		blue_led: led_2 {
+			gpios = <&gpioi 8 GPIO_ACTIVE_LOW>;
+			label = "USER LED2";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+
+		user_button: button {
+			label = "User";
+			gpios = <&gpioh 4 GPIO_ACTIVE_HIGH>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+	};
+
+	aliases {
+		led0 = &red_led;
+		led1 = &blue_led;
+		sw0 = &user_button;
+		touch = &gt911;
+		sdhc0 = &sdmmc1;
+		sram-ext = &sdram1;
+		rs232 = &usart6;
+		rs485 = &uart5;
+		eth = &mac;
+		mdio = &mdio;
+	};
+};
+
+&clk_hse {
+	clock-frequency = <DT_FREQ_M(25)>;
+	status = "okay";
+};
+
+&clk_lse {
+	status = "disabled";
+};
+
+&clk_hsi48 {
+	status = "okay";
+};
+
+&pll {
+	div-m = <5>;   /* 25/5 */
+	mul-n = <192>; /* 5*192=960 */
+	div-p = <2>;   /* 960/2=480MHz */
+	div-q = <4>;   /* 960/4=240MHz */
+	div-r = <4>;   /* 960/4=240MHz */
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&pll2 {
+	div-m = <5>;   /* 25/5 */
+	mul-n = <160>; /* 5*160=800 */
+	div-p = <4>;   /* 800/4=200MHz */
+	div-q = <4>;   /* 800/4=200MHz */
+	div-r = <4>;   /* 800/4=200MHz */
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&pll3 {
+	div-m = <5>;   /* 25/5 */
+	mul-n = <100>; /* 5*100=500 */
+	div-p = <2>;   /* 500/2=250MHz */
+	div-q = <20>;  /* 500/20=25MHz */
+	div-r = <20>;  /* 500/20=25MHz */
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;  /* 25/5*192/2 = 480 */
+	clock-frequency = <DT_FREQ_M(480)>;
+	d1cpre = <1>;  /* 480/1 = 480MHz  CPU Clocks */
+	hpre = <2>;    /* 480/2 = 240MHz */
+	d1ppre = <2>;  /* 240/2 = 120MHz APB3 Clocks */
+	d2ppre1 = <2>; /* 240/2 = 120MHz APB1 Clocks */
+	d2ppre2 = <2>; /* 240/2 = 120MHz APB2 Clocks */
+	d3ppre = <2>;  /* 240/2 = 120MHz APB4 Clocks */
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* Flash has 128KB sector size */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(128)>;
+		};
+	};
+};
+
+&uart4 {
+	pinctrl-0 = <&uart4_tx_pa0 &uart4_rx_pi9>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+	label = "debug";
+};
+
+&uart5 {
+	pinctrl-0 = <&uart5_tx_pb13 &uart5_rx_pb12>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+	label = "rs485";
+};
+
+&usart6 {
+	pinctrl-0 = <&usart6_tx_pc6 &usart6_rx_pc7>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+	label = "rs232";
+};
+
+&sdmmc1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 =
+		<&sdmmc1_d0_pc8
+		 &sdmmc1_d1_pc9
+		 &sdmmc1_d2_pc10
+		 &sdmmc1_d3_pc11
+		 &sdmmc1_ck_pc12
+		 &sdmmc1_cmd_pd2>;
+	clocks = <&rcc STM32_CLOCK(AHB3, 16)>,
+			<&rcc STM32_SRC_PLL2_R SDMMC_SEL(0)>;
+	bus-width = <4>;
+	cd-gpios = <&gpiod 5 (GPIO_PULL_UP|GPIO_ACTIVE_HIGH)>;
+	disk-name = "SD";
+	resets;
+	reset-names = "sdmmc1_reset";
+	clk-div = <4>;
+};
+
+&ltdc {
+	pinctrl-0 =
+		/* R[7:0] */
```

Comment:
these comments like this will go to garbage when file is formatted by formatting tool